### ISSUE TITLE
probes are fired in the wrong order

### ISSUE THREAD

**Issue by tyroguru** [iovisor/bpftrace#588]
If multiple probes are attached to the same probe site they are fired in reverse order which is incorrect:

```
# cat order.bt
interval:s:5
{
  printf("1\n");
}

interval:s:5
{
  printf("2\n");
}

interval:s:5
{ 
  printf("3\n");
}

#  /tmp/bpftrace ./order.bt
Attaching 3 probes...
3
2
1
```

Looks like the code in `BPFtrace::run()` that aims to correct the supposed reverse firing behaviour or bpf isn't doing what it thinks it's doing.

**Comment by mmarchini** [comment]
Apparently this affects `interval`, `tracepoint`, `profile`(?) and `hardware`. `kprobe`, `uprobe` (and therefore `usdt`) and `software` are firing in the right order:

```
# ./src/bpftrace -e 'interval:s:5 { printf("1\n") } interval:s:5 { printf("2\n") }'
Attaching 2 probes...
2
1
^C

# ./src/bpftrace -e 'k:sys_execve { printf("1\n") } k:sys_execve { printf("2\n") }'
Attaching 2 probes...
1
2
^C

# ./src/bpftrace -e 'u:bash:readline { printf("1\n"); } u:bash:readline { printf("2\n") }'
Attaching 2 probes...
1
2
^C

# ./src/bpftrace -e 't:syscalls:sys_enter_execve { printf("1\n"); } t:syscalls:sys_enter_execve { printf("2\n") }'
Attaching 2 probes...
2
1
^C

# ./src/bpftrace -e 'profile:hz:1 { printf("1\n"); } profile:hz:1 { printf("2\n") }'
Attaching 2 probes...
2
2
2
2
2
2
2
2
1
1
1
1
1
1
1
1
^C

# ./src/bpftrace -e 'software:page-faults:500 { printf("1\n"); } software:page-faults:500 { printf("2\n") }'
Attaching 2 probes...
1
2
1
2
1
2
^C

# ./src/bpftrace -e 'hardware:cache-misses:5000000 { printf("1\n"); } hardware:cache-misses:5000000 { printf("2\n") }'
Attaching 2 probes...
2
1
2
1
2
1
2
1
2
1
2
1
^C

```

**Comment by ajor** [comment]
Fixed in #627
